src/audio/module_adapter/module_adapter.c

```diff
@@ -677,6 +677,8 @@ int module_adapter_copy(struct comp_dev *dev)
 	}
 
 	if (mod->simple_copy) {
+		if(!source_c)
```

Comment:
@pjdobrowolski Could you please laborate the "certain situations" part? And even when this is true isnt it a bit too late to be checking for this? Can we check this on line 617?

Reply:
Ideally, filing an issue and then linking the PR would be the best way to help with reviews.

Reply:
@ranj063 this issue is a product of static analysis tool. I've checked your suggestions but I'm afraid that that is only solution.  
@kv2019i That PR is for private release branch for mtl and there was no need of opening issue.